x-pack/plugin/profiler/src/test/java/org/elasticsearch/xpack/profiler/GetProfilingRequestTests.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.profiler;
+
+import org.elasticsearch.common.ParsingException;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.search.SearchModule;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xcontent.NamedXContentRegistry;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static java.util.Collections.emptyList;
+
+public class GetProfilingRequestTests extends ESTestCase {
```

Comment:
You could probably have extended `AbstractXContentSerializingTestCase` ?

Reply:
`GetProfilingRequestTests`  tests different scenarios, e.g. how the parser behaves when a parse error occurs and I'm not certain how that would be handled when inheriting from `AbstractXContentSerializingTestCase`. My understanding is that this only supports a single scenario and I'd probably have to implement one test class per scenario? 

I've also implemented XContent handling "asymmetrically", in the sense that `GetProfilingRequest` only supports parsing (but not rendering an object representation to XContent) and `GetProfilingResponse` only supports rendering but not parsing. Is it common to implement both directions although only one is needed?

Reply:
oh I see what you mean, I think that's to do with the high-level client needs, which are now no longer a concern. It may be good to have bi-direction only for testing purposes, but it's overkill I guess.

Reply:
One thing I was thinking is that you could also reuse only the Wire part then, and extend AbstractWireSerliazingTestCase if it helps saving boiler plate and possibly extend the coverage too, but up to you, not a requirement.

Reply:
Thanks for the suggestion! `GetProfilingResponseTests` was a rather good candidate to inherit from `AbstractWireSerializingTestCase` so I've implemented this in https://github.com/elastic/elasticsearch/pull/91640/commits/84f1acfe8a2133dc833b56d2c7a16d19a227907d.